[assess-P0] test_geometry_hypothesis.py fails collection due to hypothesis import error

**Assessment Finding ID:** D-P0-001
**Criterion:** D — Testing
**Priority:** P0
**File:** `tests/unit/shared/test_geometry_hypothesis.py:6`

## Problem
pytest --collect-only fails with:
```
ImportError while importing test module '...test_geometry_hypothesis.py'
E   ImportError: cannot import name 'given' from 'hypothesis' (unknown location)
```

This blocks full test suite execution in environments where hypothesis is not installed.

## Evidence
- pyproject.toml lists hypothesis>=6.151.13 in dev extras
- The import from hypothesis import given, settings at line 6 fails in a standard Python 3.12 venv without hypothesis

## Recommended Fix
Either:
1. Ensure hypothesis is installed in all test environments (CI already does via .[dev]), OR
2. Skip the module gracefully when hypothesis is missing:
   ```python
   hypothesis = pytest.importorskip('hypothesis')
   ```

--- a/tests/unit/shared/test_geometry_hypothesis.py
+++ b/tests/unit/shared/test_geometry_hypothesis.py
@@ -3,6 +3,10 @@
 from __future__ import annotations
 
 import numpy as np
+import pytest
+
+pytest.importorskip("hypothesis")
+
 from hypothesis import given, settings
 from hypothesis import strategies as st
 

